joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/utils/PackageTable.scala

```diff
@@ -1,23 +1,39 @@
 package io.joern.rubysrc2cpg.utils
 
+import java.io.File as JFile
+import java.util.regex.Matcher
 import scala.collection.mutable
 
 case class MethodTableModel(methodName: String, parentClassPath: String, classType: String)
-
+case class ModuleModel(name: String, fullName: String)
+case class TypeDeclModel(name: String, fullName: String)
 case class PackageContext(moduleName: String, packageTable: PackageTable)
 
-class PackageTable() {
+class PackageTable {
 
-  val methodTableMap   = mutable.HashMap[String, mutable.HashSet[MethodTableModel]]()
-  val gemModuleMapping = mutable.HashMap[String, mutable.HashSet[String]]()
+  val methodTableMap  = mutable.HashMap[String, mutable.HashSet[MethodTableModel]]()
+  val moduleMapping   = mutable.HashMap[String, mutable.HashSet[ModuleModel]]()
+  val typeDeclMapping = mutable.HashMap[String, mutable.HashSet[TypeDeclModel]]()
 
   def addPackageMethod(moduleName: String, methodName: String, parentClassPath: String, classType: String): Unit = {
     val packageMethod = MethodTableModel(methodName, parentClassPath, classType)
     methodTableMap.getOrElseUpdate(moduleName, mutable.HashSet.empty[MethodTableModel]) += packageMethod
   }
 
-  def addModule(gemName: String, moduleName: String) = {
-    gemModuleMapping.getOrElseUpdate(gemName, mutable.HashSet.empty[String]) += moduleName
+  def addModule(gemOrFileName: String, moduleName: String, modulePath: String): Unit = {
+    val fName = gemOrFileName.split(Matcher.quoteReplacement(JFile.separator)).lastOption.getOrElse(gemOrFileName)
+    moduleMapping.getOrElseUpdate(gemOrFileName, mutable.HashSet.empty[ModuleModel]) += ModuleModel(
+      moduleName,
+      s"$fName::program.$modulePath"
+    )
+  }
+
+  def addTypeDecl(gemOrFileName: String, typeDeclName: String, typeDeclPath: String): Unit = {
+    val fName = gemOrFileName.split(Matcher.quoteReplacement(JFile.separator)).lastOption.getOrElse(gemOrFileName)
```

Comment:
I think this `.split` operation with `JFile.separator` is failing on windows. I have used `Pattern.quote(JFile.separator)` instead of `Matcher.quoteReplacement(JFile.separator)`.

Reply:
Interesting, Let me try this